### PR TITLE
Added test for #10805 Zero Dynamic Table

### LINKED ISSUE #10805: Jetty response with an invalid HTTP2 packet if the client set the hpack table size as 0
**Jetty version(s)**

Jetty 12.0.1

**Jetty Environment**

ee10

**Java version/vendor** `(use: java -version)`
17.0.8.1

**OS type/version**
suse-150400.3.30.1-x8664, Linux kernel is 5.14.21-150400.24.63-default

**Description**

Our client application is sending the http2 request with hpack table size 0, and Jetty cannot response an valid http2 packet.

``` java
// In the java org.eclipse.jetty.http2.hpack.HpackEncoder
    public HpackEncoder()
    {
       _context = new HpackContext(0); // It should set the value from 0 to HpackContext.DEFAULT_MAX_TABLE_CAPACITY
       _debug = LOG.isDebugEnabled();
        setMaxTableCapacity(HpackContext.DEFAULT_MAX_TABLE_CAPACITY);
       setTableCapacity(HpackContext.DEFAULT_MAX_TABLE_CAPACITY);
    }
```

**How to reproduce?**
In our http2 client, set the hpack table size as 0 when sending HTTP2 setting to the Jetty server. And then we get an invalid http2 response.

### PR DESCRIPTION
**Pull request by gregw**
For #10805

### CODE REVIEW

File: jetty-http2/http2-client/src/test/java/org/eclipse/jetty/http2/client/DynamicTableTest.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+import org.eclipse.jetty.http2.frames.SettingsFrame;
+import org.eclipse.jetty.http2.server.AbstractHTTP2ServerConnectionFactory;
+import org.eclipse.jetty.util.Promise;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class DynamicTableTest extends AbstractTest
```
Comment (sbordet) [inline jetty-http2/http2-client/src/test/java/org/eclipse/jetty/http2/client/DynamicTableTest.java:42]: Can you merge these tests to the same name class in the tests package?
No point having 2 separated ones.

Ah sorry -- I thought this was 12.

**Review by sbordet — CHANGES_REQUESTED** [review]

File: jetty-http2/http2-client/src/test/java/org/eclipse/jetty/http2/client/DynamicTableTest.java (hunk 1 above)
Comment (gregw) [reply jetty-http2/http2-client/src/test/java/org/eclipse/jetty/http2/client/DynamicTableTest.java:42]: @sbordet I don't understand?

**Review by sbordet — APPROVED** [review]
Is this class a port of what's in 12?
If not, then it'll need to be ported to 12, but in the different test module, and merge it with the existing one.

**Comment by gregw** [comment]
> Is this class a port of what's in 12? If not, then it'll need to be ported to 12, but in the different test module, and merge it with the existing one.

It is the same test from 12.   I did add a variant of doing both the encoder and decoder at once, but essentially the same.

**Pull request merged**